src/PostgREST/Client.hs

```diff
@@ -0,0 +1,99 @@
+        where
+          getURL :: IO Text
+          getURL =
+            -- Here, we have three cases:
+            -- 1. If the admin port config is not defined, we exit
+            --      with "no admin server error"
+            -- 2. Otherwise, if admin server is running, then we check if
+            --      postgrest server-host is configured with special hostname like "*4",
+            --      if it is, we fail with "no special hostname allowed with "--ready".
+            --      The reason for this is that we can't know the actual address.
+            -- 3. Finally, if we know the "actual" hostname and the port, then we
+            --      construct the URL and return it.
+            case configAdminServerPort of
+              Nothing   -> printAndExitWithFailure $ clientErrorMsg NoAdminServer
+              Just port ->
+                if isSpecialHostName configAdminServerHost
+                  then printAndExitWithFailure $ clientErrorMsg (NoSpecialHostNamesAllowed configAdminServerHost)
+                  else return $ makeReadyUrl port
+
+          -- NOTE: http-client automatically resolves hostnames
+          makeReadyUrl :: Int -> Text
+          makeReadyUrl p = "http://" <> wrapIfIpv6 configAdminServerHost <> ":" <> (T.pack . show) p <> "/ready"
+            where
+              -- IPv6 needs to wrapped in [], it has ':' as separator
+              wrapIfIpv6 :: Text -> Text
+              wrapIfIpv6 s
+                | T.any (== ':') s = "[" <> s <> "]"
+                | otherwise = s
+
+
+-- | Handle HTTP exception for "http-client" requests
+handleHttpException :: HttpException -> IO a
+handleHttpException (HttpExceptionRequest req _) = do
+  let url = show (HC.getUri req)
+  printAndExitWithFailure $ clientErrorMsg (HTTPConnectionRefused $ T.pack url)
+handleHttpException (InvalidUrlException url _) = do
+  printAndExitWithFailure $ clientErrorMsg (HTTPExceptionInvalidURL $ T.pack url)
+
+-- | Print the message on stdout and exit with success
+printAndExitWithSuccess :: Text -> IO a
+printAndExitWithSuccess msg = putStrLn (T.unpack msg) >> hFlush stdout >> exitSuccess
+
+-- | Print the message on stderr and exit with failure
+printAndExitWithFailure :: Text -> IO a
+printAndExitWithFailure msg = hPutStrLn stderr (T.unpack msg) >> hFlush stderr >> exitWith (ExitFailure 1)
+
+-- | ClientError to error message
+clientErrorMsg :: ClientError -> Text
+clientErrorMsg NoAdminServer                 = "UNREACHABLE: admin server is not running"
+clientErrorMsg (NoSpecialHostNamesAllowed host)
+  = "REFUSED: The --ready flag doesn't work with special hostnames like \"" <> host <> "\"."
+  <> " Alternatively, use the /ready endpoint on the admin server."
+clientErrorMsg (PostgRESTNotReady url)       = "UNAVAILABLE: " <> url
+clientErrorMsg (HTTPConnectionRefused url)   = "HTTP EXCEPTION: connection refused to " <> url
+clientErrorMsg (HTTPExceptionInvalidURL url) = "HTTP EXCEPTION: invalid url - " <> url
```

Comment:
@steve-chavez @wolfgangwalther I haven't put too much thought into the error messages, any feedback/improvements?

Reply:
Those LGTM. Just had an observation on a particular message ([on the other thread](https://github.com/PostgREST/postgrest/pull/4269/files#r2352983536))